plutus-tx-plugin/src/PlutusTx/Compiler/Builtins.hs

```diff
@@ -364,7 +367,23 @@ defineBuiltinTerms = do
     for_ enumerate $ \fun ->
         let defineBuiltinInl impl = defineBuiltinTerm annMayInline impl $ mkBuiltin fun
         in case fun of
-            PLC.IfThenElse -> defineBuiltinInl 'Builtins.ifThenElse
+            PLC.IfThenElse -> case datatypeStyle of
+                PIR.ScottEncoding  -> defineBuiltinInl 'Builtins.ifThenElse
+                PIR.SumsOfProducts -> defineBuiltinTerm annMayInline 'Builtins.ifThenElse $
+                    fmap (const annMayInline) . runQuote $ do
+                        a <- freshTyName "a"
+                        b <- freshName "b"
+                        x <- freshName "x"
+                        y <- freshName "y"
+                        return
+                            . PIR.tyAbs () a (PLC.Type ())
+                            . PIR.lamAbs () b (PLC.mkTyBuiltin @_ @Bool ())
+                            . PIR.lamAbs () x (PLC.TyVar () a)
+                            . PIR.lamAbs () y (PLC.TyVar () a)
+                            $ PIR.kase ()
+                                (PLC.TyVar () a)
+                                (PIR.Var () b)
+                                [PIR.Var () y, PIR.Var () x]
```

Comment:
Main change part 4. I'm using `datatypeStyle` here, because that's what determines whether we have `Case` available or not. I'll document it if we're fine with this logic.